### PR TITLE
add documentation to ndx-nirs

### PR DESCRIPTION
**Pull request by sumner15**
This is a working draft for collaborative code review. This PR will be converted from a draft to a PR when the documentation is complete.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1,140 @@
...
+    # for the source and detector parameters, pass in the index of
+    # the desired source (detector) in the sources (detectors) table
+    channels.add_row(label=f"S1.D1.{wavelength:.0f}nm", source=0,
+                     detector=0, source_wavelength=wavelength)
+    channels.add_row(label=f"S1.D2.{wavelength:.0f}nm", source=0,
+                     detector=1, source_wavelength=wavelength)
+    channels.add_row(label=f"S2.D1.{wavelength:.0f}nm", source=1,
+                     detector=0, source_wavelength=wavelength)
+    channels.add_row(label=f"S2.D2.{wavelength:.0f}nm", source=1,
+                     detector=1, source_wavelength=wavelength)
```
Comment (sumner15) [inline README.md]: `source_wavelength` will be correct but on this branch we've only defined the `wavelength` parameter (we haven't  updated `wavelength` -> `source_wavelength` and `emission_wavelength` on this branch) Should we merge  `enh/add-additional-snirf-fields` into `enh/add-documentation` (or equivalent merge `main` here once we merge PR #3)?

Comment (dsleiter) [reply README.md]: Just merged in main after accepting #3, so we should be all set.

File: README.md [hunk 2]
```diff
@@ -1,10 +1,223 @@
...
+            source_wavelength=wavelength,
+        )
+
+
+# create a NIRSDevice which contains all of the information
+# about the device configuration and arrangement
+device = NIRSDevice(
+    name="nirs_device",
+    description="world's best fNIRS device",
+    manufacturer="skynet",
```
Comment (dsleiter) [inline README.md:188]: :rofl:

File: README.md [hunk 3]
```diff
@@ -1,10 +1,223 @@
...
+2. Frequency Domain
+    - see `NIRSDevice.nirs_mode` and `NIRSDevice.frequency`
+3. Time-Domain 
+    - see `NIRSDevice.nirs_mode`, `NIRSDevice.time_delay`, and `NIRSDevice.time_delay_width`
+4. Diffuse Correlation Spectroscopy
+    - see `NIRSDevice.nirs_mode`, `NIRSDevice.correlation_time_delay`, and `NIRSDevice.correlation_time_delay_width`
+
+In addition, it includes support for fluorescent versions of each of these techniques.
+
+We acknowledge that NIRS is a fast-growing recording method with new modalities constantly under development. For this reason, it is possible to define other useful parameters using the `NIRSDevice.additional_parameters` field. Future version of NWB NIRS will add native support for new NIRS modalities.
```
Comment (dsleiter) [inline README.md]:
```suggestion
Other NIRS modalities are supported implicitly. We acknowledge that NIRS is a fast-growing recording method with new modalities constantly under development. For this reason, it is possible to define other useful parameters using the `NIRSDevice.additional_parameters` field. Future version of NWB NIRS will add native support for new NIRS modalities.
```

File: README.md [hunk 4]
```diff
@@ -1,10 +1,223 @@
...
+
+Integration of NIRS into the NWB data standard affords all NIRS users interoperability with many of the data storage, processing, analysis, and visualization tools already integrated within NWB. 
+
+## Modes of NIRS currently supported
+
+This extenstion currently explicitly supports: 
+
+1. Continuous Wave
+    - see `NIRSDevice.nirs_mode` 
+2. Frequency Domain
```
Comment (dsleiter) [inline README.md]:
```suggestion
2. Frequency-Domain
```

File: README.md [hunk 5]
```diff
@@ -1,10 +1,223 @@
...
+    - see `NIRSDevice.nirs_mode` and `NIRSDevice.frequency`
+3. Time-Domain 
+    - see `NIRSDevice.nirs_mode`, `NIRSDevice.time_delay`, and `NIRSDevice.time_delay_width`
+4. Diffuse Correlation Spectroscopy
+    - see `NIRSDevice.nirs_mode`, `NIRSDevice.correlation_time_delay`, and `NIRSDevice.correlation_time_delay_width`
+
+In addition, it includes support for fluorescent versions of each of these techniques.
```
Comment (dsleiter) [inline README.md:30]:
```suggestion
In addition, it includes support for fluorescent versions of each of these techniques.
  - see `NIRSChannelsTable.emssion_wavelength`
```

File: README.md [hunk 6]
```diff
@@ -1,10 +1,223 @@
...
+    - ``label`` - the label of the detector
+    - ``x``, ``y``, and ``z`` - the coordinates of the optical detector (``z`` is optional)
+
+3.  ``NIRSChannelsTable`` stores rows for each physiological channel, which is defined by source-detector pairs, where sources & detectors are referenced via ``NIRSSourcesTable`` and ``NIRSDetectorsTable``. ``NIRSChannelsTable`` includes:
+    - ``label`` - the label of the channel
+    - ``source`` - a reference to the optical source in ``NIRSSourcesTable``
+    - ``detector`` - a reference to the optical detector in ``NIRSDetectorsTable``
+    - ``source_wavelength`` - the wavelength of light in nm emitted by the source for this channel
+    - ``emission_wavelength`` - the wavelength of light in nm emitted by the fluorophone (optional; only used for fluorescent spectroscopy)
+    - ``source_power`` - the power of the source in mW used for this channel
```
Comment (dsleiter) [inline README.md]:
```suggestion
    - ``source_power`` - the power of the source in mW used for this channel (optional)
```

File: README.md [hunk 7]
```diff
@@ -1,10 +1,223 @@
...
+    - ``detector_gain`` - the gain applied to the detector for this channel
```
Comment (dsleiter) [inline README.md]:
```suggestion
    - ``detector_gain`` - the gain applied to the detector for this channel (optional)
```

File: README.md [hunk 8]
```diff
@@ -1,10 +1,223 @@
...
+    
+4. ``NIRSDevice`` defines the NIRS device itself and includes:
+    - ``channels`` - a table of the optical channels available on this device (references ``NIRSChannelsTable``)
+    - ``sources`` - the optical sources of this device (references ``NIRSSourcesTable``)
+    - ``detectors`` - the optical detectors of this device (references ``NIRSDetectorsTable``)
+    
+    ``NIRSDevice`` also includes several attributes:
+    - ``nirs_mode`` - the mode of NIRS measurement performed with this device (e.g., 'continuous-wave', 'frequency-domain', etc.)
```
Comment (dsleiter) [inline README.md]:
```suggestion
4. ``NIRSDevice`` defines the NIRS device itself and includes:
    - ``channels`` - a table of the optical channels available on this device (references ``NIRSChannelsTable``)
    - ``sources`` - the optical sources of this device (references ``NIRSSourcesTable``)
    - ``detectors`` - the optical detectors of this device (references ``NIRSDetectorsTable``)
    - ``nirs_mode`` - the mode of NIRS measurement performed with this device (e.g., 'continuous-wave', 'frequency-domain', etc.)
        
    ``NIRSDevice`` also includes several optional attributes to be used in parallel with specific ``nirs_mode`` values:
```

File: README.md [hunk 9]
```diff
@@ -1,10 +1,223 @@
...
+    - ``frequency`` - the modulation frequency in Hz for frequency domain NIRS (optional)
+    - ``time_delay`` - the time delay in ns used for gated time domain NIRS (TD-NIRS) (optional)
+    - ``time_delay_width`` - the time delay width in ns used for gated time domain NIRS (optional)
+    - ``correlation_time_delay`` - the correlation time delay in ns for diffuse correlation spectroscopy NIRS (optional)
+    - ``correlation_time_delay_width`` - the correlation time delay width in ns for diffuse correlation spectroscopy NIRS (optional)
+    - ``additional_parameters`` - any additional parameters corresponding to the NIRS device/mode that are useful for interpreting the data
```
Comment (dsleiter) [inline README.md]:
```suggestion
    - ``additional_parameters`` - any additional parameters corresponding to the NIRS device/mode that are useful for interpreting the data (optional)
```

File: docs/source/credits.rst [hunk 10]
```diff
@@ -2,20 +2,19 @@
...
 ===============
-
+AE Studio
 
 Authors
 =======
-
+Sumner L Norman
+Darin Erat Sleiter
+José Ribeiro.
```
Comment (dsleiter) [inline docs/source/credits.rst]:
```suggestion
José Ribeiro
```

File: docs/source/description.rst [hunk 11]
```diff
@@ -1,5 +1,41 @@
...
+    - ``label`` - the label of the detector
+    - ``x``, ``y``, and ``z`` - the coordinates of the optical detector (``z`` is optional)
+
+3.  ``NIRSChannelsTable`` stores rows for each physiological channel, which is defined by source-detector pairs, where sources & detectors are referenced via ``NIRSSourcesTable`` and ``NIRSDetectorsTable``. ``NIRSChannelsTable`` includes:
+    - ``label`` - the label of the channel
+    - ``source`` - a reference to the optical source in ``NIRSSourcesTable``
+    - ``detector`` - a reference to the optical detector in ``NIRSDetectorsTable``
+    - ``source_wavelength`` - the wavelength of light in nm emitted by the source for this channel
+    - ``emission_wavelength`` - the wavelength of light in nm emitted by the fluorophone (optional; only used for fluorescent spectroscopy)
+    - ``source_power`` - the power of the source in mW used for this channel
```
Comment (dsleiter) [inline docs/source/description.rst]:
```suggestion
    - ``source_power`` - the power of the source in mW used for this channel (optional)
```

File: docs/source/description.rst [hunk 12]
```diff
@@ -1,5 +1,41 @@
...
+    - ``detector_gain`` - the gain applied to the detector for this channel
```
Comment (dsleiter) [inline docs/source/description.rst]:
```suggestion
    - ``detector_gain`` - the gain applied to the detector for this channel (optional)
```

File: docs/source/description.rst [hunk 13]
```diff
@@ -1,5 +1,41 @@
...
+    
+4. ``NIRSDevice`` defines the NIRS device itself and includes:
+    - ``channels`` - a table of the optical channels available on this device (references ``NIRSChannelsTable``)
+    - ``sources`` - the optical sources of this device (references ``NIRSSourcesTable``)
+    - ``detectors`` - the optical detectors of this device (references ``NIRSDetectorsTable``)
+    
+    ``NIRSDevice`` also includes several attributes:
+    - ``nirs_mode`` - the mode of NIRS measurement performed with this device (e.g., 'continuous-wave', 'frequency-domain', etc.)
```
Comment (dsleiter) [inline docs/source/description.rst]:
```suggestion
4. ``NIRSDevice`` defines the NIRS device itself and includes:
    - ``channels`` - a table of the optical channels available on this device (references ``NIRSChannelsTable``)
    - ``sources`` - the optical sources of this device (references ``NIRSSourcesTable``)
    - ``detectors`` - the optical detectors of this device (references ``NIRSDetectorsTable``)
    - ``nirs_mode`` - the mode of NIRS measurement performed with this device (e.g., 'continuous-wave', 'frequency-domain', etc.)
        
    ``NIRSDevice`` also includes several optional attributes to be used in parallel with specific ``nirs_mode`` values:
```

File: LICENSE.txt [hunk 14]
```diff
@@ -0,0 +1,11 @@
+Copyright 2021 AE Studio
```
Comment (dsleiter) [inline LICENSE.txt:1]: :+1:

File: README.md [hunk 15]
```diff
@@ -1,10 +1,223 @@
...
+3. Time-Domain 
+    - see `NIRSDevice.nirs_mode`, `NIRSDevice.time_delay`, and `NIRSDevice.time_delay_width`
+4. Diffuse Correlation Spectroscopy
+    - see `NIRSDevice.nirs_mode`, `NIRSDevice.correlation_time_delay`, and `NIRSDevice.correlation_time_delay_width`
+
+In addition, it includes support for fluorescent versions of each of these techniques.
+
+We acknowledge that NIRS is a fast-growing recording method with new modalities constantly under development. For this reason, it is possible to define other useful parameters using the `NIRSDevice.additional_parameters` field. Future version of NWB NIRS will add native support for new NIRS modalities.
+
+## Other data standards
```
Comment (dsleiter) [inline README.md]:
```suggestion
## Related data standards 
```

File: README.md [hunk 16]
```diff
@@ -1,10 +1,223 @@
...
+            detector=i_detector,
+            source_wavelength=wavelength,
+        )
+
+
+# create a NIRSDevice which contains all of the information
+# about the device configuration and arrangement
+device = NIRSDevice(
+    name="nirs_device",
+    description="world's best fNIRS device",
```
Comment (dsleiter) [inline README.md:187]: :100:

**Review by dsleiter — APPROVED** [review]
I think this is looking pretty good, ready for a first version. Nice work! I pushed some additions, and made some suggestions for some modifications which you can feel free to accept or reject. After reviewing the additions & changes, unless there are any major remaining questions or items to discuss, feel free to merge in the morning!

**Comment by dsleiter** [comment]
Resolves #6

**Pull request merged**